twake/backend/node/src/services/workspaces/services/workspace.ts

```diff
@@ -773,4 +784,78 @@ export class WorkspaceServiceImpl implements TwakeServiceProvider, Initializable
       }
     }
   }
+
+  /**
+   * Sets the invitation domain for the specified workspace
+   *
+   * @param {{ company_id: string; workspace_id: string }} pk - the primary key
+   * @param {String} domain - domain
+   * @param {ExecutionContext} context - the execution context
+   */
+  setInviteDomain = async (
+    pk: { company_id: string; workspace_id: string },
+    domain: string,
+    context?: ExecutionContext,
+  ): Promise<void> => {
+    const workspace = await gr.services.workspaces.get(
+      {
+        company_id: pk.company_id,
+        id: pk.workspace_id,
+      },
+      context,
+    );
+
+    if (!workspace) {
+      logger.error("failed to set invitation domain: workspace not found");
+      throw CrudException.notFound("Workspace entity not found");
+    }
+
+    workspace.preferences = {
+      invite_domain: domain,
+    };
```

Comment:
If there was a workspace.preferences.invite_domain before and was different, you need to remove the record from workspaceInviteDomainRepository first